internal/secrets/localsecrets/localsecrets.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package localsecrets provides a way to encrypt and decrypt small messages
+// without making network calls to a third party service.
+package localsecrets
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"golang.org/x/crypto/nacl/secretbox"
+	"io"
+)
+
+// secretKeeper holds a secret for use in symmetric encryption,
+// and implementations of driver.Encryper and driver.Decrypter.
+type secretKeeper struct {
+	secretKey [32]byte // secretbox key size
+	Encrypter driver.Encrypter
+	Decrypter driver.Decrypter
+}
+
+// NewSecretKeeper takes a secret key and returns a secretKeeper.
+func NewSecretKeeper(sk string) *secretKeeper {
```

Comment:
It's generally a mistake to return an unexported type from an exported function, as it means that no one calling the function can express the return type (i.e. pass the type around).

Reply:
WDYT about having two separate constructors entirely? E.g., `NewEncrypter` and `NewDecrypter`? They would return the corresponding concrete type, but could both use the same struct `secretKeeper` underneath (since it would implement both `driver` interfaces).

Reply:
see answer at: https://github.com/google/go-cloud/pull/899/files#r239657534